tree-diff

```diff
@@ -1,10 +1,22 @@
 #!/usr/bin/env python3
 
 import argparse
+import hashlib
 import json
 import os
 
 
+def hash_file(fd):
+    BLOCK_SIZE = 65536
+    hasher = hashlib.sha256()
+    buf = os.read(fd, BLOCK_SIZE)
+    while len(buf) > 0:
+        hasher.update(buf)
+        buf = os.read(fd, BLOCK_SIZE)
+
+    return f"sha256-{hasher.hexdigest()}"
```

Comment:
`sha512:` is the usual prefix, rather than `sha512-`.

Reply:
Changed.